lib/ui/src/components/preview/iframe.js

```diff
@@ -12,20 +11,20 @@ export class IFrame extends Component {
   }
 
   shouldComponentUpdate(nextProps) {
-    const { scale, src } = this.props;
-    return scale !== nextProps.scale || src !== nextProps.src;
-  }
-
-  componentDidUpdate(prevProps) {
     const { scale } = this.props;
-    if (scale !== prevProps.scale) {
+
+    if (scale !== nextProps.scale) {
       this.setIframeBodyStyle({
-        width: `${scale * 100}%`,
-        height: `${scale * 100}%`,
-        transform: `scale(${1 / scale})`,
+        width: `${nextProps.scale * 100}%`,
+        height: `${nextProps.scale * 100}%`,
+        transform: `scale(${1 / nextProps.scale})`,
         transformOrigin: 'top left',
       });
     }
+
+    // this component renders an iframe, which gets updates via post-messages
```

Comment:
@ndelangen Where is the handler code for the post-messages?

Reply:
In the Preview component, this is just the Iframe component which is a part of the Preview component.